packages/downloader/download.py

```
@@ -0,0 +1,126 @@
+                    progress_data = json.load(ro)
+                progress_data["status"] = "finished"
+                
+                # Write atomically
+                temp_path = OUTPUTS_DIR / f'{info_dict["id"]}.json.tmp'
+                with open(temp_path, 'w') as wo:
+                    json.dump(progress_data, wo)
+                os.rename(temp_path, progress_file)
+            except Exception:
+                pass
+
+def info(url: str):
```

Ruff (0.12.2)

Missing return type annotation for public function `info`

(ANN201)